### DISCUSSION TITLE
From last saved cursor?

### DISCUSSION THREAD

**Discussion by jobqueue** [tweepy/tweepy#605] [Questions] [answered]
```
        cursor = tweepy.Cursor(api.followers_ids, screen_name=handle)
        for page in cursor.pages():
                         process(page)
```

In case if process failed, I'd like to save something about the last succeeded cursor/page position and continue the operation from the cursor position, is that possible with Tweepy?

**Comment by natejlong** [comment]
```
tweepy_cursor = tweepy.Cursor(self.api.followers_ids, screen_name=handle).pages()
save_obj = {}
while True:
    try:
         page = tweepy_cursor.next()
         cursor = tweepy_cursor.next_cursor
         for follower in page:
               writer.writerow([follower])
     except StopIteration:
         break

     # update and write save file in case we need to resume later
     save_obj['cursor'] = cursor
     self.write_save_file(save_obj=save_obj)
```

**Comment by Harmon758** [comment, marked as answer]
If the pagination is by cursor, as is the case with [`API.followers_ids`](https://tweepy.readthedocs.io/en/latest/api.html#API.followers_ids), then the iterator you're using is a CursorIterator returned by `Cursor.pages`:
https://github.com/tweepy/tweepy/blob/8ce6e8ce2591f3d14547851552524c624e1373a3/tweepy/cursor.py#L11-L31  In which case, you can use the `prev_cursor` and `next_cursor` attributes of the CursorIterator:
https://github.com/tweepy/tweepy/blob/8ce6e8ce2591f3d14547851552524c624e1373a3/tweepy/cursor.py#L61-L68
